Userland/Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -62,4 +94,544 @@ void HTMLMediaElement::pause() const
+        // FIXME: 14. ⌛ If the node after pointer is a source element, let candidate be that element.
+        // FIXME: 15. ⌛ Advance pointer so that the node before pointer is now the node that was after pointer, and the node after pointer is the node after
+        //            the node that used to be after pointer, if any.
+        // FIXME: 16. ⌛ If candidate is null, jump back to the search loop step. Otherwise, jump back to the process candidate step.
+        // FIXME: 17. ⌛ Waiting: Set the element's networkState attribute to the NETWORK_NO_SOURCE value.
+        // FIXME: 18. ⌛ Set the element's show poster flag to true.
+        // FIXME: 19. ⌛ Queue a media element task given the media element to set the element's delaying-the-load-event flag to false. This stops delaying the
+        //            load event.
+        // FIXME: 20. End the synchronous section, continuing the remaining steps in parallel.
+        // FIXME: 21. Wait until the node after pointer is a node other than the end of the list. (This step might wait forever.)
+        // FIXME: 22. Await a stable state. The synchronous section consists of all the remaining steps of this algorithm until the algorithm says the
+        //            synchronous section has ended. (Steps in synchronous sections are marked with ⌛.)
+        // FIXME: 23. ⌛ Set the element's delaying-the-load-event flag back to true (this delays the load event again, in case it hasn't been fired yet).
+        // FIXME: 24. ⌛ Set the networkState back to NETWORK_LOADING.
+        // FIXME: 25. ⌛ Jump back to the find next candidate step above.
+        break;
+    }
+
+    return {};
+}
+
+enum class FetchMode {
+    Local,
+    Remote,
+};
+
+// https://html.spec.whatwg.org/multipage/media.html#concept-media-load-resource
+WebIDL::ExceptionOr<void> HTMLMediaElement::fetch_resource(AK::URL const& url_record, Function<void()> failure_callback)
+{
+    auto& realm = this->realm();
+    auto& vm = realm.vm();
+
+    // 1. If the algorithm was invoked with media provider object or a URL record whose blob URL entry is a blob URL entry whose object is a media provider
+    //    object, then let mode be local. Otherwise let mode be remote.
+    // FIXME: Detect media provider object / blob URLs with a media provider object.
+    auto mode = FetchMode::Remote;
+
+    // FIXME: 2. If mode is remote, then let the current media resource be the resource given by the URL record passed to this algorithm; otherwise, let the
+    //           current media resource be the resource given by the media provider object. Either way, the current media resource is now the element's media
+    //           resource.
+    // FIXME: 3. Remove all media-resource-specific text tracks from the media element's list of pending text tracks, if any.
+
+    // 4. Run the appropriate steps from the following list:
+    switch (mode) {
+        // -> If mode is remote
```

Comment:
Comment indentation is off, I hope clang-format didn't do this

Reply:
clang-format accepts either indentation above a `case` label so pushing them back one tab is manual :( missed this one